src/test.rs

```diff
@@ -892,12 +907,12 @@ impl Debug for TestCase {
 }
 
 impl TestCase {
-    pub fn new(commit: Commit, test: Arc<Test>) -> Self {
-        Self {
-            cache_hash: test.cache_policy.cache_hash(&commit),
+    pub async fn new<W: Worktree + ?Sized>(commit: Commit, test: Arc<Test>, worktree: &W) -> anyhow::Result<Self> {
```

Comment:
This is another place where it might be better to preload the hash to avoid async getting everywhere.